.github/workflows/prover-testing.yml

```diff
@@ -13,18 +13,19 @@ jobs:
     runs-on: [self-hosted, ubuntu-20.04, X64, small]
     name: Prover static check
     steps:
-    - name: install Go
-      uses: actions/setup-go@v5
-      with:
-        go-version: 1.23.x
     - name: checkout code
       uses: actions/checkout@v4
       with:
         fetch-depth: 0
-    - uses: actions/cache@v4
+    - name: install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: 1.23.x
+        cache-dependency-path: |
+              prover/go.sum
+    - uses: actions/cache@v4.2.0
```

Comment:
Why bump to v4.2.0, I think v4 already point to v4.2.0: https://github.com/actions/cache/tags

Reply:
In the #github-actions-self-hosted-runners Slack channel, there's a recommendation from DevOps to pin the Github Action version so that they are the same in every run.

They gave an example that `gradle/actions/setup-gradle@v4` should be `gradle/actions/setup-gradle@cc4fc85e6b35bafd578d5ffbc76a5518407e1af0 #v4.2.1` instead

Reply:
Actually I'm working on a [PR](https://github.com/Consensys/linea-monorepo/pull/584) for actions/cache, given that it was not used properly (see [issue](https://github.com/Consensys/linea-monorepo/issues/589)), and we might even drop it if there's no performance gain (i.e. it seems the integration and e2e tests would be running longer when cache is used for pulling docker images)